testcases/two_zone_apartment_hydronic/models/TwoZoneApartmentHydronic/Components/Rooms/thermalZone.mo

```diff
@@ -0,0 +1,859 @@
+    use_m_flow_in=true,
+    use_C_in=true,
+    nPorts=1) "source model for air infiltration"
+    annotation (Placement(transformation(extent={{8,0},{16,8}})));
+  Modelica.Blocks.Math.Gain gain(k=-1)
+    annotation (Placement(transformation(extent={{-16,2},{-8,10}})));
+  Modelica.Fluid.Interfaces.FluidPort_a supplyAir(redeclare final package
+      Medium = MediumA) "Supply air"
+    annotation (Placement(transformation(extent={{-110,22},{-90,42}}),
+        iconTransformation(extent={{-110,22},{-90,42}})));
+  Modelica.Fluid.Interfaces.FluidPort_b returnAir(redeclare final package
+      Medium = MediumA) "Return air"
+    annotation (Placement(transformation(extent={{-110,-30},{-90,-10}}),
+        iconTransformation(extent={{-110,-30},{-90,-10}})));
+  TwoZoneApartmentHydronic.Components.BaseClasses.InternalLoad lig(
+    radFraction=0.5,
+    latPower_nominal=0,
+    senPower_nominal=1.5)
+    annotation (Placement(transformation(extent={{-100,48},{-88,60}})));
+  TwoZoneApartmentHydronic.Components.BaseClasses.InternalLoad equ(
+    latPower_nominal=0,
+    senPower_nominal=4,
+    radFraction=0.5)
+    annotation (Placement(transformation(extent={{-100,72},{-88,84}})));
+  TwoZoneApartmentHydronic.Components.BaseClasses.OccupancyLoad occ(
+    radFraction=0.5,
+    co2Gen=8.64e-6,
+    occ_density=1/Afloor,
+    senPower=60,
+    latPower=20)
+    annotation (Placement(transformation(extent={{-100,60},{-88,72}})));
+  Modelica.Blocks.Math.MultiSum sumRad(nu=3) "Sum of radiant internal gains"
+    annotation (Placement(transformation(extent={{-48,76},{-42,82}})));
+  Modelica.Blocks.Math.MultiSum sumCon(nu=3) "Sum of convective internal gains"
+    annotation (Placement(transformation(extent={{-50,66},{-44,72}})));
+  Modelica.Blocks.Math.MultiSum sumLat(nu=3) "Sum of latent internal gains"
+    annotation (Placement(transformation(extent={{-48,56},{-42,62}})));
+  IBPSA.Utilities.IO.SignalExchange.Read reaPLig(y(unit="W"), description=
+        "Lighting power submeter",
+    KPIs=IBPSA.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.ElectricPower)
+                                   "Read lighting power consumption"
+    annotation (Placement(transformation(extent={{-26,46},{-18,54}})));
+  IBPSA.Utilities.IO.SignalExchange.Read reaPPlu(y(unit="W"), description=
+        "Plug load power submeter",
+    KPIs=IBPSA.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.ElectricPower)
+                                    "Read plug load power consumption"
+    annotation (Placement(transformation(extent={{-26,88},{-18,96}})));
+  Modelica.Blocks.Math.MultiSum sumLig(k=fill(roo.AFlo, 2), nu=2)
+    "Lighting power consumption"
+    annotation (Placement(transformation(extent={{-48,46},{-42,52}})));
+  Modelica.Blocks.Math.MultiSum sumPlu(k=fill(roo.AFlo, 2), nu=2)
+    "Plug power consumption"
+    annotation (Placement(transformation(extent={{-48,88},{-42,94}})));
+  IBPSA.Utilities.IO.SignalExchange.Read reaTRooAir(

```

Comment:
It would be better to use signal exchange blocks from Buildings or IDEAS rather than rely on the IBPSA library itself.  This could then remove the reliance on the IBPSA library in the library_versions.json.

Reply:
Removed all dependencies from IBPSA library